contracts/PetPark.sol

```diff
@@ -1,7 +1,124 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.0;
 
-
 contract PetPark {
+    address owner;
+
+    enum AnimalType {
+        None,
+        Fish,
+        Cat,
+        Dog,
+        Rabbit,
+        Parrot
+    }
+    enum Gender {
+        Male,
+        Female
+    }
+
+    struct User {
+        AnimalType animal;
+        Gender gender;
+        uint age;
+    }
+
+    mapping(AnimalType => uint) public animalCounts;
+    mapping(address => User) public userInfo;
+
+    event Added(AnimalType animal, uint count);
+    event Borrowed(AnimalType animal);
+    event Returned(AnimalType animal);
+
+    constructor() {
+        owner = msg.sender;
+    }
+
+    modifier onlyOwner() {
+        require(owner == msg.sender, "Not an owner");
+        _;
+    }
+
+    modifier isOldEnough(uint age) {
+        require(age != 0, "Invalid Age");
+        _;
+    }
+
+    modifier isValidAnimal(AnimalType animal) {
+        require(animal != AnimalType.None, "Invalid animal");
+        _;
+    }
+
+    modifier isValidAnimalType(AnimalType animal) {
+        require(animal != AnimalType.None, "Invalid animal type");
+        _;
+    }
+
+    modifier isAvailableAnimal(AnimalType animal) {
+        require(animalCounts[animal] > 0, "Selected animal not available");
+        _;
+    }
+
+    function add(AnimalType _animal, uint _count)
+        external
+        onlyOwner
+        isValidAnimal(_animal)
+    {
+        animalCounts[_animal] = animalCounts[_animal] + _count;
+        emit Added(_animal, animalCounts[_animal]);
+    }
+
+    function borrow(
+        uint _age,
+        Gender _gender,
+        AnimalType _animal
+    )
+        external
+        isOldEnough(_age)
+        isValidAnimalType(_animal)
+        isAvailableAnimal(_animal)
+    {
+        User storage u = userInfo[msg.sender];
+
+        if (u.age != _age && u.age != 0) {
```

Comment:
The age, gender and animal checks for the case when the pet was already borrowed, should be in a single if clause. Once you move them, all your tests will pass.

Reply:
My bad.I meant `if block` not `if clause`. Something like this 

```
if (u.age != 0) { // If clause to check if the pet was already borrowed
  // All the conditions go here
}
```